clang/test/compiler-course/chistov_a_user_data_type/chistov_a_user_data_type_test.cpp

```diff
@@ -0,0 +1,88 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/PrintData_Chistov_Alexey_FIIT1_ClangAST%pluginext -plugin PrintDataPlugin %s -fsyntax-only 2>&1 | FileCheck %s
+
+class A {};
+// CHECK: A(class)
+
+struct B {};
+// CHECK: B(struct)
+
+struct Simple {
+    int x;
+    double y;
+
+    int method(int x, int y) {}
+};
+
+// CHECK: Simple(struct)
+// CHECK-NEXT: |_Fields
+// CHECK-NEXT: | |_ x (int|public)
+// CHECK-NEXT: | |_ y (double|public)
+// CHECK-NEXT: |_Methods
+// CHECK-NEXT: | |_ method (int|public)
+
+template <typename T>
+class Template {
+    T x;
+    int z;
+};
+
+// CHECK: Template(class|template)
+// CHECK-NEXT: |_Fields
+// CHECK-NEXT: | |_ x (T|private)
+// CHECK-NEXT: | |_ z (int|private)
```

Comment:
Slightly different order
```suggestion
// CHECK: Template(class|template)
// CHECK-NEXT: |_Fields
// CHECK-NEXT: | |_ x (T|private)
// CHECK-NEXT: | |_ z (int|private)
template <typename T>
class Template {
    T x;
    int z;
};
```